Src/POI.DiscordDotNet/Services/Implementations/DiscordStarboardService.cs

```diff
@@ -0,0 +1,128 @@
+			return;
+		}
+
+		// Skip event if the message is in the starboard channel (To prevent people staring the bot messages)
+		if(args.Channel.Id == serverSettings.StarboardChannelId)
+		{
+			return;
+		}
+
+		// Check if the message reactions contains the star emote
+		if (args.Message.Reactions.All(x => x.Emoji.Name != "⭐"))
+		{
+			return;
+		}
+
+		var messageStarCount = args.Message.Reactions.First(x => x.Emoji.Name == "⭐").Count;
+		// Check if the message has enough stars
+		if (messageStarCount < serverSettings.StarboardEmojiCount)
+		{
+			return;
+		}
+
+		// Get the starboard channel by the server settings id
+		var starboardChannel = await sender.GetChannelAsync(serverSettings.StarboardChannelId.Value);
+		if (starboardChannel == null)
+		{
+			_logger.LogError("Starboard channel not found!");
+			return;
+		}
+
+		// Get DiscordMember from DiscordUser to get the display name
+		var author = await args.Guild.GetMemberAsync(args.Message.Author.Id);
+		if (author == null)
+		{
+			_logger.LogError("Author not found!");
+			return;
+		}
+
+		// Get the starboard message from the database
+		var foundMessage = await _starboardMessagesRepository.FindOneByServerIdAndChannelIdAndMessageId(args.Guild.Id, args.Channel.Id, args.Message.Id);
+		// If the message is already in the database, update the star count
+		// (This will also update the message contents)
+		var embed = GetStarboardEmbed(author.DisplayName, args.Message.Channel.Name, args.Message.Content, args.Message.JumpLink, args.Message.Timestamp, (uint)messageStarCount, args.Message.Attachments.FirstOrDefault()?.Url);
+
+		if(foundMessage != null)
+		{
+			var starboardMessage = await starboardChannel.GetMessageAsync(foundMessage.StarboardMessageId);
+			await starboardMessage.ModifyAsync(msg => msg.Embed = embed);
+			_logger.LogInformation("Updated message {JumpLink} with {stars} stars!", args.Message.JumpLink, messageStarCount);
+			return;
+		}
+
+		// If the message is not in the database, create a new starboard message
+		var embedMessage = await starboardChannel.SendMessageAsync(embed);
+		// And add to the database.
+		await _starboardMessagesRepository.Insert(new StarboardMessages(args.Guild.Id, args.Channel.Id, args.Message.Id, embedMessage.Id));
+		_logger.LogInformation("Message {JumpLink} sent to starboard channel!", args.Message.JumpLink);
+	}
+
+	private static DiscordEmbed GetStarboardEmbed(string userName, string channelName, string content, Uri jumpLink, DateTimeOffset timestamp, uint messageStarCount, string? attachmentUrl = null)
+	{
+		var builder = new DiscordEmbedBuilder()
+			.WithTitle($"{userName} in #{channelName}")
+			.WithDescription(content)
+			// .WithDescription(content.Length > 50 ? content[..50] + "..." : content)
+			.WithColor(new DiscordColor(0x6CCAF1))
+			.WithUrl(jumpLink)
+			.WithFooter($"⭐{messageStarCount}")
+			.WithTimestamp(TimeZoneInfo.ConvertTimeBySystemTimeZoneId(timestamp, "Europe/Brussels"));
```

Comment:
Not sure if this is really needed as the system timezone for POI is already configured to be dependent on ENV variables?